native/src/seal/encryptor.cpp

```diff
@@ -173,8 +191,32 @@ namespace seal
     }
 
     void Encryptor::encrypt_internal(
-        const Plaintext &plain, bool is_asymmetric, bool save_seed, Ciphertext &destination,
-        MemoryPoolHandle pool) const
+        const Plaintext &plain, 
+        bool is_asymmetric, 
+        bool save_seed, 
+        Ciphertext &destination,
+        MemoryPoolHandle pool
+    ) const
+    {
+        // Do not export noise.
+        StaticPolynomialArray u_destination;
```

Comment:
A common pattern in crypto libraries is to zero the memory on destruction. I need to find out the threat model to determine if that makes sense for us.

Reply:
Other things like ciphertext don't have a destructor for that explicitly, although it is possible that they zero out the data through the underlying DynArray

Reply:
Added a constructor that by default zeros out the data on destruction, and added a boolean to change that behavior should we want to later.

I looked more into the `DynArray` implementation that backs `Ciphertext` in SEAL, but it is not obvious if the data is zeroed out in SEAL. Essentially the `DynArray` will call all the underlying deconstructors but I am not precisely sure which ones are getting called at a brief pass. We can look more deeply later if we want to ensure that ciphertexts and plaintexts zero out on deallocation.